cdap-examples/SpamClassifier/src/test/java/co/cask/cdap/examples/sparkstreaming/SpamClassifierTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.examples.sparkstreaming;
+
+import co.cask.cdap.common.utils.Networks;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.ServiceManager;
+import co.cask.cdap.test.SparkManager;
+import co.cask.cdap.test.StreamManager;
+import co.cask.cdap.test.TestBase;
+import co.cask.common.http.HttpRequest;
+import co.cask.common.http.HttpRequests;
+import co.cask.common.http.HttpResponse;
+import com.google.common.base.Charsets;
+import com.google.common.io.Closeables;
+import org.apache.twill.internal.kafka.EmbeddedKafkaServer;
+import org.apache.twill.internal.kafka.client.ZKKafkaClientService;
+import org.apache.twill.internal.zookeeper.InMemoryZKServer;
+import org.apache.twill.kafka.client.Compression;
+import org.apache.twill.kafka.client.KafkaClientService;
+import org.apache.twill.kafka.client.KafkaPublisher;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test for {@link SpamClassifier}
+ */
+public class SpamClassifierTest extends TestBase {
+
+  private static final String KAFKA_TOPIC = "someTopic";
+  private static final String KAFKA_BROKER_ID = "1";
+  private static ZKClientService zkClient;
+  private static KafkaClientService kafkaClient;
+
+  private static InMemoryZKServer zkServer;
+  private static EmbeddedKafkaServer kafkaServer;
+  private static int kafkaPort;
+
+  @BeforeClass
+  public static void initialize() throws Exception {
+    TestBase.initialize();
```

Comment:
Better to use a different method name so that you don't need to call parent. The parent `@BeforeClass` will get called first.